scripts/stats.js

```diff
@@ -62,32 +63,60 @@ for (let i = 0; i < layers.length; i++) {
 }
 
 if (opts.layerCount) {
-  console.log(`${layerCount} layers`);
+  if (opts.json) {
+    process.stdout.write(JSON.stringify({ layerCount }) + "\n");
+  } else {
+    console.log(`${layerCount} layers`);
+  }
 }
 
 if (opts.layerSize) {
-  console.log(`Total layer size ${layerSize.toLocaleString("en-US")} bytes`);
+  if (opts.json) {
+    process.stdout.write(JSON.stringify({ layerSize }) + "\n");
+  } else {
+    console.log(`Total layer size ${layerSize.toLocaleString("en-US")} bytes`);
+  }
 }
 
 if (opts.allLayers) {
-  console.log(`${layerCount} layers, grouped as follows:`);
-  layerGroupSizeStats.forEach((v, k) => {
-    let layerCount = layerGroupCountStats.get(k);
-    let layerString = `${k}(${layerCount})`.padEnd(30, ".");
-    console.log(
-      `${layerString}${v.toLocaleString("en-US").padStart(10, ".")} bytes`
-    );
-  });
+  if (opts.json) {
+    let allLayers = {
+      layerCount,
+      layers: {},
+    };
+    layerGroupSizeStats.forEach((v, k) => {
+      let layerCount = layerGroupCountStats.get(k);
+      allLayers.layers[k] = layerCount;
+    });
+    process.stdout.write(JSON.stringify(allLayers) + "\n");
+  } else {
+    layerGroupSizeStats.forEach((v, k) => {
+      let layerCount = layerGroupCountStats.get(k);
+      let layerString = `${k}(${layerCount})`.padEnd(30, ".");
+      console.log(
+        `${layerString}${v.toLocaleString("en-US").padStart(10, ".")} bytes`
+      );
+    });
+  }
 }
 
 if (opts.printGroup) {
-  layerGroupMap.get(opts.printGroup).forEach((lyr) => console.log(lyr));
+  let group = layerGroupMap.get(opts.printGroup);
+  if (opts.json) {
+    let layers = [];
+    group.forEach((lyr) => layers.push(lyr));
+    process.stdout.write(JSON.stringify(layers) + "\n");
+  } else {
+    group.forEach((lyr) => console.log(lyr));
+  }
 }
 
 if (opts.printLayer) {
   if (opts.pretty) {
-    console.log(JSON.stringify(layerMap.get(opts.printLayer), null, 2));
+    process.stdout.write(
+      JSON.stringify(layerMap.get(opts.printLayer), null, 2) + "\n"
```

Comment:
This output is more like JSON-LD. Anyone using this output programmatically will need to extract an arbitrary line to work with it. Let’s save these bits in a single variable scoped to the whole function and spit it out at the end.